### DISCUSSION TITLE
Política empeora con el entrenamiento despues de cierto punto

### DISCUSSION THREAD

**Discussion by furcelay** [JavierBZ/Tarea3-IIC3697#7] [Q&A] [answered]
Hola!

En el entrenamiento del modelo el reward promedio por episodio mejoró mucho durante las primeras 6 épocas pero después comenzó a disminuir y al final se estancó en un valor mucho más bajo ¿Es normal que ocurra eso? ¿A qué podría deberse? 

Me imagino que tiene relación con que los labels para el entrenamiento son dinámicos entonces la política puede cambiar de forma drástica al actualizar el modelo target...

**Comment by JavierBZ** [comment, marked as answer]
Hola, es normal que pueda pasar eso. En general puede deberse a las nuevas experiencias que debe aprender. Recuerda que el juego tiene distintos ambientes, parte de día, pero después puede toparse con la noche o niebla que dificultan la visibilidad, o suelo congelado que dificulta el control del auto. Puede pasar que la red intenta superar estas experiencias más difíciles, pero no lo logra optimamente y eso también afecta lo que aprendió en experiencias más fáciles (de día), ya que son los mismos pesos que debe entrenar para distintos escenarios.

Esto es una hipótesis mía y es posible que no sea completamente verdad jaja, pero al menos explica la dificultad de aprender en ambientes cambiantes. Lo que tu dices también puede ser verdad y se complementa con lo que expliqué.

Si quieres mejorar ese problema te sugiero ajustar la tasa de aprendizaje, ya sea con un valor constante menor o con un planificador o _scheduler_  de pytorch. Revisa la [documentación](https://pytorch.org/docs/stable/optim.html) de los optimizadores para más detalle de estos últimos.

Espero te sirva!

**Comment by furcelay** [comment]
Muchas gracias!!

Me hace mucho sentido lo que me dices, es mucho más complejo el problema si tiene que adaptarse a un ambiente cambiante. Voy a revisar lo de los planificadores.

Saludos!
